gpx.go

```diff
@@ -171,86 +161,55 @@ type MovingData struct {
 	MaxSpeed        float64
 }
 
-// SpeedsAndDistances represents speed and distance
-type SpeedsAndDistances struct {
-	Speed    float64
-	Distance float64
-}
-
-// LocationsResultPair represents a result from location query
-type LocationsResultPair struct {
-	SegmentNo int
-	PointNo   int
+// speedsAndDistances represents speed and distance
+type speedsAndDistances struct {
+	speed    float64
+	distance float64
 }
 
 /*==========================================================*/
 
-// Parse parses a GPX file and return a Gpx object.
-func Parse(gpxPath string) (*Gpx, error) {
-	gpxFile, err := os.Open(gpxPath)
+// Parse parses a GPX reader and return a Gpx object.
+func Parse(r io.Reader) (*Gpx, error) {
+	g := NewGpx()
+	d := xml.NewDecoder(r)
+	err := d.Decode(g)
 	if err != nil {
-		// fmt.Println("Error opening file: ", err)
-		return nil, err
+		return nil, fmt.Errorf("couldn't parse gpx data: %v", err)
 	}
-	defer gpxFile.Close()
-
-	b, err := ioutil.ReadAll(gpxFile)
+	return g, nil
+}
 
+// ParseFile reads a GPX file and parses it.
+func ParseFile(filepath string) (*Gpx, error) {
+	f, err := os.Open(filepath)
 	if err != nil {
-		// fmt.Println("Error reading file: ", err)
 		return nil, err
 	}
-	g := NewGpx()
-	xml.Unmarshal(b, &g)
-
-	return g, nil
+	return Parse(f)
 }
 
 /*==========================================================*/
 
-func getTime(timestr string) time.Time {
-	t, err := time.Parse(timelayout, timestr)
-	if err != nil {
-		return time.Time{}
-	}
-	return t
-}
-
 func toXML(n interface{}) []byte {
 	content, _ := xml.MarshalIndent(n, "", "	")
 	return content
 }
 
-func getMinimaMaximaStart() *Bounds {
-	return &Bounds{
-		MaxLat: -math.MaxFloat64,
-		MinLat: math.MaxFloat64,
-		MaxLon: -math.MaxFloat64,
-		MinLon: math.MaxFloat64,
-	}
+var maxBounds = Bounds{
```

Comment:
I can't remember anymore but I think I took a function exactly because there are no const structs in go. It's a while ago...

I kind of like the function better than the var. Myabe you could reintroduce `getMaxBounds()`